utils/rpms/daos.spec

```diff
@@ -32,7 +32,7 @@ BuildRequires: python36-scons >= 2.4
 BuildRequires: scons >= 2.4
 %endif
 BuildRequires: libfabric-devel >= %{libfabric_version}
-BuildRequires: mercury-devel >= %{mercury_version}
+BuildRequires: mercury-devel = %{mercury_version}
```

Comment:
Why does this need to be changed to `=` from `>=`?  This will hamper our ability to update mercury for this branch without also have to re-qualify and entire DAOS release (which is expensive -- which is why we like to allow ourselves to update components without having to create a whole new DAOS release).

Reply:
We have updated Mercury for master to utilize a new feature in UCX available only in recent versions of the UCX distribution.  We are not yet ready to the enable the required UCX (MOFED) versions in CI, so it was decided by technical leadership to have us pin the prior Mercury version for the 2.2 DAOS release.

Reply:
> We are not yet ready to the enable the required UCX (MOFED) versions in CI

I'm not sure I understand this statement.  CI uses the same MOFED version on all branches (master, release/2.*).  I'm not sure what _enable_ means in this context.

Reply:
As much as the concern is pinning and creating a lot more work for us should we need to make a new mercury release for 2.2, the bigger concern is that if mercury@master cannot be used for 2.2 any more, how do we create an updated mercury for only 2.2.x should we need to make a bugfix release, etc. in the future?

By wanting to pin here, you are effectively declaring that you want to fork mercury because by needing to pin, you are declaring that mercury@master is no longer supportable on release/2.2.  By declaring that. you are also declaring that any future release of mercury for 2.2 will need to be forked from the commit in mercury that produced `mercury-2.2.0-1` and not master.

Given that mercury@master is still 2.2.0, at release -3, what would the version of a mercury forked from 2.2.0-1 be versioned?  I suppose it could be 2.2.0-1.1 but I'm not sure how well that works with the `%{dist}` that is in the package `Release:`.  I.e. I am not sure if you can say `BuildRequires: mercury-devel >= 2.2.0-1.1%{?dist}, mercury-devel < 2.2.0-2%{?dist}` and get the results you would expect.  That would require some experimentation.  It's not typical to start creating pinning ranges based on the `Release:` value in all of my experience.

Moreover, however, I am not sure our existing infrastructure can support creating branches in dependencies.  Maybe it's not as complicated as it might seem and maybe it "just works", but I don't know one way or the other.  Being able to have forked dependencies has never been identified as a requirement.

In any case, I suppose we can just pin here now and cross the bridges and fight the fires that might come up in the future when they do.  I just fear we might be painting ourselves into a corner we cannot get ourselves out of without considering the future now, before we make a commitment here.

Reply:
well, we will just unpin when we release 2.2.1 and upgrade Mercury. I don't think that we  will deliver any Mercury update to 2.2.0 anyways.

Reply:
> we will just unpin when we release 2.2.1 and upgrade Mercury

Ahhh.  So DAOS 2.2.1 will be compatible with the current mercury@master?  If so, that is good but was not made clear 
previously.  It still leaves us in a tough position for 2.2.0 though should we need to fix a mercury bug.

> I don't think that we will deliver any Mercury update to 2.2.0 anyways.

The problem is that we cannot predict that.  We cannot know with certainty that there isn't an undiscovered bug in a piece of software.  Indeed, if we could accurately predict that software will never have any bugs our jobs would be much much easier.  :-)